tests/impl.cpp

```diff
@@ -5649,7 +5649,16 @@ result_t test_mm_stream_si32(const SSE2NEONTestImpl &impl, uint32_t i)
 
 result_t test_mm_stream_si64(const SSE2NEONTestImpl &impl, uint32_t i)
 {
-    return TEST_UNIMPL;
+    const int64_t a = (const int64_t) impl.mTestInts[i];
+#if defined(__x86_64__) || defined(__i386__)
+    long long int p[1];
```

Comment:
Why using `long long int` here?
I thought `int64_t` is better.

Reply:
It seems the function argument on x86 is in `long long int*` as I present below, so I am using two different datatype here
```c
_mm_stream_si64(long long *__p, long long __a)
{
  __builtin_ia32_movnti64(__p, __a);
}
```

Reply:
Where did you find this?
It seems like the document is still using the `__int64*`, which can be replaced with `int64_t*`.
https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_stream_si64

Reply:
I guess it depends on the compiler?
I was using `int64_t` on x86 on node1, but I failed to compile the source code

Reply:
What is the error message?

Reply:
the error message is as following 

```
tests/impl.cpp:5659:21: error: invalid conversion from ‘int64_t*’ {aka ‘long int*’} to ‘long long int*’ [-fpermissive]
 5659 |     _mm_stream_si64(p, a);
      |                     ^
      |                     |
      |                     int64_t* {aka long int*}
In file included from tests/common.h:7,
                 from tests/impl.h:3,
                 from tests/impl.cpp:1:
/usr/lib/gcc/x86_64-linux-gnu/9/include/emmintrin.h:1477:33: note:   initializing argument 1 of ‘void _mm_stream_si64(long long int*, long long int)’
 1477 | _mm_stream_si64 (long long int *__A, long long int __B)
      |                  ~~~~~~~~~~~~~~~^~~
make: *** [Makefile:48: tests/impl.o] Error 1
```

Reply:
here is the unit test for the above case 
```c
result_t test_mm_stream_si64(const SSE2NEONTestImpl &impl, uint32_t i)
{
    const int64_t a = (const int64_t) impl.mTestInts[i];
    int64_t p[1];
    _mm_stream_si64(p, a);
    ASSERT_RETURN(p[0] == a);
    return TEST_SUCCESS;
}
```

Reply:
I thought using `int64_t*` directly should be workable since `int64_t` and `long long int` are integers with 64-bit size.

But taking `stdint.h` of **GCC** as an example, it only says that `long int` is equal to `int64_t` when the architecture is 64-bit.
We should treat `int64_t(long int)` and `long long int` differently since they are different data type.
If the casting is added like the below code, the error would be eliminated.
```c++
_mm_stream_si64((long long int*)p, a);
```

The [__mm_stream_si64()](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_stream_si64) document uses `__int64` as the argument type, which is a data type used in **Microsoft Visual C++** and equals to `long long int`.
As I searched for `__mm_stream_si64()` definition in `emmintrin.h` header file, it turns out that the argument type is `long long int` in **GCC** and **Clang**.

I think we just need to add the type casting without extra `x86` detection.

@jserv do you have any suggestion?

Reply:
It is a matter of [64-bit data models](https://en.wikipedia.org/wiki/64-bit_computing#64-bit_data_models): Microsoft Windows (x86-64) takes LLP64 while Most Unix and Unix-like systems take LP64.

Reply:
I think we should do the casting to avoid architecture detection?

Reply:
what kinds of casting does you mean?

Reply:
This one:
```c
_mm_stream_si64((long long int*)p, a);
```